lib/mongo/cluster.rb

```diff
@@ -89,14 +86,15 @@ def add(host)
     # @since 2.0.0
     def initialize(seeds, options = {})
       @addresses = []
+      @addresses_update = Mutex.new
       @servers = []
+      @servers_update = Mutex.new
```

Comment:
Thanks for your review.
I'm not going to take this approach because there is IO in a bunch of places that might not be obvious (like in the instantiation of an Address, for example) and the code would get messy if I was frequently grabbing and releasing the lock in between lines of code in Cluster.

I have taken your suggestion to only use one lock for both the servers and addresses list.

You're also right about having to copy the list. I've added that in as well.

Thanks again!